v3/src/components/case-table/use-index-column.tsx

```diff
@@ -59,10 +59,48 @@ export const IndexCell = ({ caseId, index, onClick }: ICellProps) => {
     // no dependencies means we'll check/fix it after every render
   })
 
+  // Find the parent CODAP component to display the index menu above the grid
+  useEffect(() => {
+    if (cellElt && !codapComponentElt) {
+      let parent: HTMLElement | null
+      for (parent = cellElt; parent; parent = parent.parentElement) {
+        if (parent.classList.contains("codap-component")) {
+          setCodapComponentElt(parent)
+          break
+        }
+      }
+    }
+  }, [cellElt, codapComponentElt])
+
+  const handleMoveDataRow = () => {
+    alert("Move Data Row Here clicked")
+  }
+
+  const handleInsertCase = () => {
+    alert("Insert Case clicked")
+  }
+
+  const handleInsertCases = () => {
+    alert("Insert Cases clicked")
+  }
+
+  const handleDeleteCase = () => {
+    alert("Delete Case clicked")
+  }
+
   return (
-    <div ref={setNodeRef} className="codap-index-content" data-testid="codap-index-content"
-      onClick={e => onClick?.(caseId, e) }>
-      {index != null ? `${index + 1}` : ""}
-    </div>
+    <Menu isLazy>
+      <MenuButton ref={setNodeRef} className="codap-index-content" data-testid="codap-index-content">
+        {index != null ? `${index + 1}` : ""}
+      </MenuButton>
+      {codapComponentElt && createPortal((
+        <MenuList>
+          <MenuItem onClick={handleMoveDataRow}>Move Data Entry Row Here</MenuItem>
+          <MenuItem onClick={handleInsertCase}>Insert Case</MenuItem>
+          <MenuItem onClick={handleInsertCases}>Insert Cases...</MenuItem>
+          <MenuItem onClick={handleDeleteCase}>Delete Case</MenuItem>
+        </MenuList>
```

Comment:
Ultimately, we're not going to want all of these handlers in the `useIndexColumn()` hook. One way to achieve that would be to create a separate `IndexMenuList` component which renders the `<MenuList>` and has the handlers so this would just be:
```suggestion
        <IndexMenuList />
```
Also, "Delete Case" should be "Delete Cases" when multiple cases are selected.